test/int/datafederation_test.go

```diff
@@ -0,0 +1,186 @@
+			By("Deleting the deployment in Atlas manually", func() {
+				// We need to remove the deployment in Atlas manually to let project get removed
+				_, err := atlasClient.AdvancedClusters.Delete(context.Background(), createdProject.ID(), createdDataFederation.Name)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(checkAtlasDeploymentRemoved(createdProject.Status.ID, createdDataFederation.Name), 600, interval).Should(BeTrue())
+				createdDataFederation = nil
+			})
+		}
+		if createdProject != nil && createdProject.Status.ID != "" {
+			if createdDataFederation != nil {
+				By("Removing Atlas DataFederation " + createdDataFederation.Name)
+				Expect(k8sClient.Delete(context.Background(), createdDataFederation)).To(Succeed())
+				deploymentName := createdDataFederation.Name
+				if customresource.ResourceShouldBeLeftInAtlas(createdDataFederation) || customresource.ReconciliationShouldBeSkipped(createdDataFederation) {
+					By("Removing Atlas DataFederation " + createdDataFederation.Name + " from Atlas manually")
+					Expect(deleteAtlasDataFederation(createdProject.Status.ID, deploymentName)).To(Succeed())
+				}
+				Eventually(checkAtlasDataFederationRemoved(createdProject.Status.ID, deploymentName), 600, interval).Should(BeTrue())
+			}
+
+			By("Removing Atlas Project " + createdProject.Status.ID)
+			Expect(k8sClient.Delete(context.Background(), createdProject)).To(Succeed())
+			Eventually(checkAtlasProjectRemoved(createdProject.Status.ID), 60, interval).Should(BeTrue())
+		}
+		removeControllersAndNamespace()
+	})
+
+	Describe("DataFederation can be created with stores and private endpoints", func() {
+		It("Should Succeed", func() {
+			By("Creating a DataFederation instance with DB and STORE", func() {
+				createdDataFederation = mdbv1.NewDataFederationInstance(createdProject.Name, dataFederationInstanceName, namespace.Name)
+				Expect(k8sClient.Create(context.Background(), createdDataFederation)).ShouldNot(HaveOccurred())
+
+				Eventually(func(g Gomega) {
+					df, _, err := dataFederationClient.Get(context.Background(), createdProject.ID(), createdDataFederation.Spec.Name)
+					g.Expect(err).ShouldNot(HaveOccurred())
+					g.Expect(df).NotTo(BeNil())
+				}).WithTimeout(20 * time.Minute).WithPolling(15 * time.Second).ShouldNot(HaveOccurred())
+			})
+
+			By("Adding a new DB and STORE", func() {
+				df := &mdbv1.AtlasDataFederation{}
+				Expect(k8sClient.Get(context.Background(), types.NamespacedName{
+					Namespace: namespace.Name,
+					Name:      dataFederationInstanceName,
+				}, df)).To(Succeed())
+
+				dfu := df.WithStorage(&mdbv1.Storage{
+					Databases: []mdbv1.Database{
+						{
+							Name: "test-db-1",
+							Collections: []mdbv1.Collection{
+								{
+									Name: "test-collection-1",
+									DataSources: []mdbv1.DataSource{
+										{
+											StoreName: "http-test",
+											Urls: []string{
+												"https://data.cityofnewyork.us/api/views/vfnx-vebw/rows.csv",
```

Comment:
I think would be good to have the CSV offline, upload it to a S3 bucket and use it on the tests, we will neve now when this would be unavailable and break the test. Can be done a improvement.